clients/unshaded/src/main/java/tachyon/client/lineage/TachyonLineage.java

```diff
@@ -37,14 +40,16 @@
  * but goes through the implementations provided in {@link AbstractLineageClient}.
  */
 @PublicApi
+@ThreadSafe
 public final class TachyonLineage extends AbstractLineageClient {
+  private static AtomicBoolean sInitialized = new AtomicBoolean(false);
   private static TachyonLineage sTachyonLineage;
 
   /**
    * @return the current lineage for Tachyon
    */
   public static synchronized TachyonLineage get() {
-    if (sTachyonLineage == null) {
+    if (!sInitialized.getAndSet(true)) {
```

Comment:
why we need to use a atomic boolean here?
isn't this method already synchronized?